code/modules/projectiles/guns/energy.dm

```diff
@@ -594,3 +594,776 @@
+	burst_amount = 4
+	fire_sound = 'sound/weapons/guns/fire/Laser Rifle Standard.ogg'
+	message_to_user = "You set the Laser Carbine's charge mode to standard fire."
+	gun_firemode_list = list(GUN_FIREMODE_AUTOMATIC, GUN_FIREMODE_BURSTFIRE, GUN_FIREMODE_AUTOBURST)
+	fire_mode = GUN_FIREMODE_BURSTFIRE
+	icon_state = "tec"
+
+/datum/energy_carbine_mode/base/spread
+	charge_cost = 60
+	ammo = /datum/ammo/energy/lasgun/marine/blast
+	fire_delay = 1.2 SECONDS
+	burst_amount = 1
+	fire_sound = 'sound/weapons/guns/fire/Laser Carbine Scatter.ogg'
+	message_to_user = "You set the Laser Carbine's charge mode to spread."
+	fire_mode = GUN_FIREMODE_SEMIAUTO
+	icon_state = "tec"
+	radial_icon_state = "laser_spread"
+
+/obj/item/weapon/gun/energy/lasgun/standard_marine_carbine/unique_action(mob/user)
+	return switch_modes(user)
+
+/obj/item/weapon/gun/energy/lasgun/standard_marine_carbine/proc/switch_modes(mob/user)
+	if(!user)
+		CRASH("switch_modes called with no user.")
+
+	var/list/available_modes = list()
+	for(var/mode in mode_list)
+		available_modes += list("[mode]" = image(icon = initial(mode_list[mode].radial_icon), icon_state = initial(mode_list[mode].radial_icon_state)))
+
+	var/datum/energy_carbine_mode/choice = mode_list[show_radial_menu(user, user, available_modes, null, 64, tooltips = TRUE)]
+	if(!choice)
+		return
+
+
+	playsound(user, 'sound/weapons/emitter.ogg', 5, FALSE, 2)
+
+
+	gun_firemode = initial(choice.fire_mode)
+
+
+	charge_cost = initial(choice.charge_cost)
+	ammo = GLOB.ammo_list[initial(choice.ammo)]
+	fire_delay = initial(choice.fire_delay)
+	burst_amount = initial(choice.burst_amount)
+	fire_sound = initial(choice.fire_sound)
+	SEND_SIGNAL(src, COMSIG_GUN_AUTOFIREDELAY_MODIFIED, fire_delay)
+	SEND_SIGNAL(src, COMSIG_GUN_BURST_SHOTS_TO_FIRE_MODIFIED, burst_amount)
+	SEND_SIGNAL(src, COMSIG_GUN_FIRE_MODE_TOGGLE, initial(choice.fire_mode), user.client)
+
+	base_gun_icon = initial(choice.icon_state)
+	update_icon()
+
+	to_chat(user, initial(choice.message_to_user))
+
+	var/obj/screen/ammo/A = user.hud_used.ammo //The ammo HUD
+	A.update_hud(user)
+
+	return TRUE
+
+
+
+//Ammo/Charge functions
+/obj/item/weapon/gun/energy/lasgun/standard_marine_carbine/update_icon(mob/user)
+	var/cell_charge = (!cell || cell.charge <= 0) ? 0 : CEILING((cell.charge / max(cell.maxcharge, 1)) * 100, 25)
+	icon_state = "[base_gun_icon]_[cell_charge]"
+	update_mag_overlay(user)
+
+
+/obj/item/weapon/gun/energy/lasgun/standard_marine_carbine/update_item_state(mob/user)
+    . = item_state
+    item_state = "[initial(icon_state)][flags_item & WIELDED ? "_w" : ""]"
+    if(. != item_state && ishuman(user))
+        var/mob/living/carbon/human/human_user = user
+        if(src == human_user.l_hand)
+            human_user.update_inv_l_hand()
+        else if (src == human_user.r_hand)
+            human_user.update_inv_r_hand()
+
+
+//TE Standard Sniper
+
+obj/item/weapon/gun/energy/lasgun/standard_marine_sniper
+	name = "\improper TE-S Sniper rifle"
```

Comment:
```suggestion
	name = "\improper TE-S laser strike rifle"
```
because mjp said no, do you think it applies to your PR?

Reply:
I don't even know what laser strike would even imply
Personally "Lasniper" or "Laser sniper rifle" would be better and fits the simple and direct naming style that we use now

Reply:
```suggestion
	name = "\improper TE-S laser sniper rifle"
```

Reply:
doing laser sniper, not strike.